datanommer.models/datanommer/models/__init__.py

```diff
@@ -128,7 +131,14 @@ def add(envelope):
     obj.msg = message['msg']
     obj.headers = headers
 
-    session.add(obj)
+    try:
+        session.add(obj)
+        session.flush()
```

Comment:
For the same reason we don't commit after every DML operation, so that everything is consistent before commit. If an error happened later on (when adding the user/package -> message mappings for example), it would throw an error, roll back the entire transaction, and you could potentially retry in the future, when the message is redelivered by the broker (depending on your message ACK settings) for example.